docs/src/prototypes/chatMessages/ThreadedMessages/ThreadReplyEditor.tsx

```diff
@@ -0,0 +1,73 @@
+import * as React from 'react'
+import * as keyboardKey from 'keyboard-key'
+
+import { Button, Flex, Input, Toolbar, Ref, Chat } from '@stardust-ui/react'
+import { toolbarItems } from './mockData'
+
+interface ThreadReplyEditorState {
+  editMode: boolean
+}
+
+class ThreadReplyEditor extends React.Component<{}, ThreadReplyEditorState> {
+  state = {
+    editMode: false,
+  }
+
+  buttonRef = React.createRef<HTMLElement>()
+  inputRef = React.createRef<HTMLElement>()
+
+  renderReplyButton = () => {
+    return (
+      <Ref innerRef={this.buttonRef}>
+        <Button
+          fluid
+          className="ui-button__reply"
+          content="Reply"
+          onClick={() => {
+            this.setState({ editMode: true }, () => {
+              if (this.inputRef && this.inputRef.current) {
+                this.inputRef.current.focus()
+              }
+            })
+          }}
+        />
+      </Ref>
+    )
+  }
+
+  renderEditor = () => {
+    return (
+      <Chat.Message className="ui-chat__message__reply-editor">
+        <Flex
+          column
+          onKeyDown={e => {
```

Comment:
Ouch. May be we can use `createComponent()` instead? As I see `editMode` is not used anywhere in `styles`

Reply:
oops, actually this `onKeyDown` should be on `Chat.Message`.
Anyways, how would `createComponent` help if currently I can't reuse `actionHandlers` functionality and create my own handlers?

Reply:
Okay, if there will be no key handlers on `Flex` and don't have any issues 🐱

Reply:
💃